clients/cobol-lsp-vscode-extension/src/services/Settings.ts

```diff
@@ -49,3 +49,97 @@ export function createFileWithGivenPath(folderPath: string, fileName: string, pa
     }
 
 }
+
+/**
+ * SettingsService provides read/write configurstion settings functionality
+ */
+export class SettingsService {
+    /**
+     * Get list of local subroutine path
+     * @returns a list of local subroutine path
+     */
+    public static getSubroutineLocalPath(): string[] {
+        return vscode.workspace.getConfiguration().get(SETTINGS_SUBROUTINE_LOCAL_KEY);
+    }
+
+    /**
+     * Get copybook local path based on program file name
+     * @param cobolProgramName is a program file name
+     * @returns a list of local path
+     */
+    public static getCopybookLocalPath(cobolProgramName: string): string[] {
+        const pathList: string[] = vscode.workspace.getConfiguration(SETTINGS_CPY_SECTION).get(PATHS_LOCAL_KEY);
+        return SettingsService.evaluateVariable(pathList, "program_name", cobolProgramName);
+    }
+
+    /**
+     * Get Lsp Port from configuration
+     * @returns lsp port number
+     */
+    public static getLspPort(): number | undefined {
+        return +vscode.workspace.getConfiguration().get(SERVER_PORT);
+    }
+
+    /**
+     * Determine if dsn path exists in the configurstion 
+     * @returns true if path exists and false otherwise
+     */
+    public static hasDsnPath(): boolean {
+        return vscode.workspace.getConfiguration(SETTINGS_CPY_SECTION).has(PATHS_ZOWE)        
+    }
+
+    /**
+     * Get list of dsn path
+     * @returns a list of dsn path
+     */
+    public static getDsnPath(): string[] {
+        return vscode.workspace.getConfiguration(SETTINGS_CPY_SECTION).get(PATHS_ZOWE);        
+    }
+
+    /**
+     * Set list of dsn path to the configuration
+     * @param paths is a list of new values
+     */
+    public static setDsnPath(paths: string[]) {
+        vscode.workspace.getConfiguration(SETTINGS_CPY_SECTION).update(PATHS_ZOWE, paths);
+    }
+
+    /**
+     * Determine if uss path exists in the configurstion 
+     * @returns true if path exists and false otherwise
+     */
+    public static hasUssPath(): boolean {
+        return vscode.workspace.getConfiguration(SETTINGS_CPY_SECTION).has(PATHS_USS)        
+    }
+
+    /**
+     * Get list of uss path
+     * @returns a list of uss path
+     */
+    public static getUssPath(): string[] {
```

Comment:
I don't see a setter for the USS path, is it intentional?

Reply:
There are no setter because it is read-only for now and setter is not needed

Reply:
For some reason we have a command for editing DSN's, but not for the rest like USS and local. I think we need to follow the same approach for all settings. So either delete the editing command or add something similar for the rest. This is not a part of the PR. We will discuss it separately.